Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM ubuntu:20.04
+RUN apt-get update
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    python3-distutils \
+    python-is-python3 \
+    python3 \
+    python3-pip \
+    g++ \
+    cmake \
+    ccache \
+    binutils-gold \
+    binutils-dev \
+    ninja-build \
+    pkg-config \
+    gcc-multilib \
+    g++-multilib \
+    gawk \
+    dos2unix \
+    libxml2-dev \
+    rsync \
+    git \
+    libtool \
+    m4 \
+    automake \
+    libgcrypt-dev \
+    liblzma-dev \
+    libssl-dev \
+    libgss-dev \
+    python3-dev \
+    wget \
+    zlib1g-dev
+COPY . /ml-compiler-opt
```

Comment:
We also need define a volume for any input and output data, see https://docs.docker.com/engine/reference/builder/#volume. This will be used for toolchain, corpus, vocab, warmstart, and the trained model.

Reply:
I defined one volume mounted at `/external`. I'm thinking it might be better to have individual volumes for all of the components listed above though? Let me know what works better for your workflow.

Reply:
I think that's fine for now, we can always add more volumes later if needed.

Reply:
Sounds good. Should be good to merge then?